pkg/apis/pipeline/v1beta1/pipeline_validation_test.go

```diff
@@ -361,10 +404,15 @@ func TestPipelineSpec_Validate_Failure(t *testing.T) {
 				TaskRef:         &TaskRef{Name: "foo-task"},
 				WhenExpressions: []WhenExpression{{}},
 			}},
+			Finally: []PipelineTask{{
+				Name:            "invalid-pipeline-task-finally",
+				TaskRef:         &TaskRef{Name: "foo-task"},
+				WhenExpressions: []WhenExpression{{}},
+			}},
 		},
 		expectedError: apis.FieldError{
 			Message: `missing field(s)`,
-			Paths:   []string{"tasks[1].when[0]"},
+			Paths:   []string{"tasks[1].when[0]", "finally[0].when[0]"},
```

Comment:
im a little nervous (which maybe is a sign that im overthinking this!) about all the test here combining the two scenarios, i.e. validation failure in the pipelinetask AND in the finally task.

at least since the path contains both we can know both were caught, but im wondering if it'd be good to also cover a case where only one is set.

since the new cases we want to add are so similar to the existing ones, we have a few choices:

1. Combine the cases together like this PR does
2. Duplicate the cases
3. To be extra sure, do (2) AND do (1) (test them all separately AND together)

I think (3) is overkill, so I'm happy to go with this - I feel like if we do (1) we should do (2) for _at least_ one scenario, or do (2) and have one test that combines the cases like in (1)

Reply:
we can't have `finally tasks` without `dag tasks`, so we can't do option 2 with `finally tasks` only

left the `dag tasks` as they are, and added cases with combined `dag tasks` and `finally tasks`

Reply:
actually ended up doing (3), but the ones with finally tasks have dag tasks as well (but the dag tasks in some of those cases don't have when expressions)